packages/block-library/src/heading/edit.js

```diff
@@ -10,33 +15,37 @@ import { __ } from '@wordpress/i18n';
 import { Fragment } from '@wordpress/element';
 import { PanelBody } from '@wordpress/components';
 import { createBlock } from '@wordpress/blocks';
+import { withSelect } from '@wordpress/data';
 import {
 	RichText,
 	BlockControls,
 	InspectorControls,
 	AlignmentToolbar,
 } from '@wordpress/block-editor';
 
-export default function HeadingEdit( {
+function HeadingEdit( {
 	attributes,
 	setAttributes,
 	mergeBlocks,
 	insertBlocksAfter,
 	onReplace,
 	className,
+	levelChoices,
 } ) {
 	const { align, content, level, placeholder } = attributes;
 	const tagName = 'h' + level;
-
+	const BlockControlsLevelsRange = ( levelChoices.length <= 3 ) ?
+		levelChoices :
+		levelChoices.slice( 1, 4 ); //
 	return (
 		<Fragment>
 			<BlockControls>
-				<HeadingToolbar minLevel={ 2 } maxLevel={ 5 } selectedLevel={ level } onChange={ ( newLevel ) => setAttributes( { level: newLevel } ) } />
+				<HeadingToolbar levelsRange={ BlockControlsLevelsRange } selectedLevel={ level } onChange={ ( newLevel ) => setAttributes( { level: newLevel } ) } />
 			</BlockControls>
 			<InspectorControls>
 				<PanelBody title={ __( 'Heading Settings' ) }>
 					<p>{ __( 'Level' ) }</p>
-					<HeadingToolbar minLevel={ 1 } maxLevel={ 7 } selectedLevel={ level } onChange={ ( newLevel ) => setAttributes( { level: newLevel } ) } />
```

Comment:
Did the default max level change from 7 to 6? And if so, why?

Reply:
See my comment here: https://github.com/WordPress/gutenberg/pull/14689/files#r271809189